lib/membrane/core/parent/children_model.ex

```diff
@@ -1,66 +1,43 @@
 defmodule Membrane.Core.Parent.ChildrenModel do
   @moduledoc false
 
-  alias Membrane.{Child, ChildEntry}
+  alias Membrane.{Child, ChildEntry, ParentError}
   alias Membrane.Core.Parent
 
   @type children_t :: %{Child.name_t() => ChildEntry.t()}
 
-  @spec add_child(Parent.state_t(), Child.name_t(), ChildEntry.t()) ::
-          {:ok | {:error, {:duplicate_child, Child.name_t()}}, Parent.state_t()}
-  def add_child(%{children: children} = state, child, data) do
-    if Map.has_key?(children, child) do
-      {{:error, {:duplicate_child, child}}, state}
-    else
-      {:ok, %{state | children: children |> Map.put(child, data)}}
-    end
-  end
-
-  @spec get_child_data(Parent.state_t(), Child.name_t()) ::
-          {:ok, child_data :: ChildEntry.t()}
-          | {:error, {:unknown_child, Child.name_t()}}
-  def get_child_data(%{children: children}, child) do
-    children[child] |> Bunch.error_if_nil({:unknown_child, child})
+  @spec assert_child_exists!(Parent.state_t(), Child.name_t()) :: :ok
+  def assert_child_exists!(state, child) do
+    _data = get_child_data!(state, child)
+    :ok
   end
 
-  @spec pop_child(Parent.state_t(), Child.name_t()) ::
-          {{:ok, child_data :: ChildEntry.t()}
-           | {:error, {:unknown_child, Child.name_t()}}, Parent.state_t()}
-  def pop_child(%{children: children} = state, child) do
-    {pid, children} = children |> Map.pop(child)
-
-    with {:ok, pid} <- pid |> Bunch.error_if_nil({:unknown_child, child}) do
-      state = %{state | children: children}
-      {{:ok, pid}, state}
+  @spec get_child_data!(Parent.state_t(), Child.name_t()) :: ChildEntry.t()
+  def get_child_data!(%{children: children}, child) do
+    case children do
+      %{^child => data} -> data
+      _children -> raise ParentError, "Unknown child #{inspect(child)}"
```

Comment:
I wonder if this message will be clear enough for the user to figure out what's wrong. Can you think of example where this might appear and how the stacktrace will look? Plus, WDYT about adding the list of children that are there?

```suggestion
      _children -> raise ParentError, "Unknown child #{inspect(child)}. Current children: #{inspect(Map.keys(children))}"
```